src/core/agent/services/authService.ts

```diff
@@ -0,0 +1,106 @@
+import { AgentService } from "./agentService";
+import { AgentServicesProps, MiscRecordId } from "../agent.types";
+import { Agent } from "../agent";
+import { LoginAttempts } from "./auth.types";
+import { BasicRecord } from "../records";
+
+class AuthService extends AgentService {
+  static readonly TIME_UNIT = 60 * 1000;
+
+  constructor(agentServiceProps: AgentServicesProps) {
+    super(agentServiceProps);
+  }
+
+  static readonly LOGIN_ATTEMPT_RECORD_NOT_FOUND =
+    "Login attempt record not found";
+
+  async getLoginAttempts() {
+    const attemptInfo = await Agent.agent.basicStorage.findById(
+      MiscRecordId.LOGIN_ATTEMPT
+    );
+
+    if (attemptInfo) {
+      return attemptInfo.content as unknown as LoginAttempts;
+    } else {
+      await Agent.agent.basicStorage.createOrUpdateBasicRecord(
```

Comment:
we can remove the `else` here and keep the rest outside if statement, since the other clause calls `return`. This is more readable